crates/evm/evm/src/executors/invariant/corpus.rs

```diff
@@ -0,0 +1,381 @@
+                        }
+                    }
+                    // interleave
+                    2 => {
+                        trace!(target: "corpus", "interleave {} with {}", primary.uuid, secondary.uuid);
+                        if should_evict {
+                            self.current_mutated = Some(primary.uuid);
+                        }
+                        for (tx1, tx2) in primary.tx_seq.iter().zip(secondary.tx_seq.iter()) {
+                            // chunks?
+                            let tx = if rng.gen_bool(0.5) { tx1.clone() } else { tx2.clone() };
+                            new_seq.push(tx);
+                        }
+                    }
+                    // 3. Overwrite prefix with new sequence.
+                    3 => {
+                        let corpus = if rng.gen_bool(0.5) { primary } else { secondary };
+                        trace!(target: "corpus", "overwrite prefix of {}", corpus.uuid);
+                        if should_evict {
+                            self.current_mutated = Some(corpus.uuid);
+                        }
+                        new_seq = corpus.tx_seq.clone();
+                        for i in 0..rng.gen_range(0..=new_seq.len()) {
+                            new_seq[i] = self.new_tx(test_runner)?;
+                        }
+                    }
+                    // 4. Overwrite suffix with new sequence.
+                    4 => {
+                        let corpus = if rng.gen_bool(0.5) { primary } else { secondary };
+                        trace!(target: "corpus", "overwrite suffix of {}", corpus.uuid);
+                        if should_evict {
+                            self.current_mutated = Some(corpus.uuid);
+                        }
+                        new_seq = corpus.tx_seq.clone();
+                        for i in
+                            new_seq.len() - rng.gen_range(0..new_seq.len())..corpus.tx_seq.len()
+                        {
+                            new_seq[i] = self.new_tx(test_runner)?;
+                        }
+                    }
+                    // 5. Select idx to mutate and change its args according to its ABI.
+                    5 => {
+                        let targets = test.targeted_contracts.targets.lock();
+                        let corpus = if rng.gen_bool(0.5) { primary } else { secondary };
+                        trace!(target: "corpus", "ABI mutate args of {}", corpus.uuid);
+                        if should_evict {
+                            self.current_mutated = Some(corpus.uuid);
+                        }
+                        new_seq = corpus.tx_seq.clone();
+
+                        let idx = rng.gen_range(0..new_seq.len());
+                        let tx = new_seq.get_mut(idx).unwrap();
+                        if let (_, Some(function)) = targets.fuzzed_artifacts(tx) {
+                            tx.call_details.calldata =
```

Comment:
This completely changes the calldata and I think it'd be good to be able to only affect one argument e.g.
deposit(AAAAA,1) -> deposit(AAAAA, 100000)

Reply:
added it here https://github.com/foundry-rs/foundry/pull/10190/commits/27dc9c72241fe00a07e84fe2976cf162b4e009fb